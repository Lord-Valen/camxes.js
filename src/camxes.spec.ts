import { describe, beforeAll, it, expect } from "vitest";
import { camxes } from "./index";

describe("parser", () => {
  let sut: any;

  beforeAll(() => {
    sut = camxes;
  });

  it("is defined", () => {
    expect(sut).toBeDefined();
    expect(sut.parse).toBeDefined();
  });

  // TODO: Implement reverse templating
  it("parses basic jafra", async () => {
    expect(sut.parse("coi")).toMatchSnapshot();

    expect(sut.parse("mi nelci")).toMatchSnapshot();
  });
});
