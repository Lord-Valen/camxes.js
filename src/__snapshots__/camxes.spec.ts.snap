// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parser > parses basic jafra 1`] = `
[
  "text",
  [
    "text_part_2",
    [
      [
        "free",
        [
          "vocative",
          [
            [
              "COI_clause",
              [
                [
                  "COI",
                  "coi",
                ],
              ],
            ],
          ],
        ],
      ],
    ],
  ],
]
`;

exports[`parser > parses basic jafra 2`] = `
[
  "text",
  [
    "text_1",
    [
      "paragraphs",
      [
        "paragraph",
        [
          "statement",
          [
            "statement_1",
            [
              "statement_2",
              [
                "statement_3",
                [
                  "sentence",
                  [
                    [
                      "terms",
                      [
                        "terms_1",
                        [
                          "terms_2",
                          [
                            "term",
                            [
                              "term_1",
                              [
                                "sumti",
                                [
                                  "sumti_1",
                                  [
                                    "sumti_2",
                                    [
                                      "sumti_3",
                                      [
                                        "sumti_4",
                                        [
                                          "sumti_5",
                                          [
                                            "sumti_6",
                                            [
                                              "KOhA_clause",
                                              [
                                                [
                                                  "KOhA",
                                                  "mi",
                                                ],
                                              ],
                                            ],
                                          ],
                                        ],
                                      ],
                                    ],
                                  ],
                                ],
                              ],
                            ],
                          ],
                        ],
                      ],
                    ],
                  ],
                  [
                    "bridi_tail",
                    [
                      "bridi_tail_1",
                      [
                        "bridi_tail_2",
                        [
                          "bridi_tail_3",
                          [
                            "selbri",
                            [
                              "selbri_1",
                              [
                                "selbri_2",
                                [
                                  "selbri_3",
                                  [
                                    "selbri_4",
                                    [
                                      "selbri_5",
                                      [
                                        "selbri_6",
                                        [
                                          "tanru_unit",
                                          [
                                            "tanru_unit_1",
                                            [
                                              "tanru_unit_2",
                                              [
                                                "BRIVLA_clause",
                                                [
                                                  [
                                                    "BRIVLA",
                                                    [
                                                      "gismu",
                                                      "nelci",
                                                    ],
                                                  ],
                                                ],
                                              ],
                                            ],
                                          ],
                                        ],
                                      ],
                                    ],
                                  ],
                                ],
                              ],
                            ],
                          ],
                        ],
                      ],
                    ],
                  ],
                ],
              ],
            ],
          ],
        ],
      ],
    ],
  ],
]
`;
